### DISCUSSION TITLE
Time interval based retention for ID partitioned table

### DISCUSSION THREAD

**Discussion by neetam15** [pgpartman/pg_partman#409] [Q&A] [answered]
My parent table has an bigint ID column and a created_date timestamp column. I am partitioning the table over the ID column (this, id-based partition). However, I want the retention based on the created_date column such that partitions older than 3years can be dropped. 
I see that only time based partitioning can have retention configured as a time-interval. 

Is there any alternative to achieve this?

**Comment by keithf4** [comment, marked as answer]
Directly built into pg_partman, there currently is no way to do this since retention is solely based on the partition column and its data type.

I do have plans to see about allowing multi-column partitioning in the future, which may allow this easier. But that is going to wait until I see the results of this discussion around trigger-based partitioning ( https://github.com/pgpartman/pg_partman/discussions/360 ). It will be much easier to refactor the code if I don't have to worry about supporting the old methods anymore.

You can however manage this yourself by setting up a process that scans for the min-max values in each table based on your created_date column and drops the tables accordingly based on the result.

**Reply by neetam15** [reply]
Thank you for the prompt response @keithf4 ! That helps.
